### ISSUE TITLE
Implement softmax and cross entropy kernels

### ISSUE THREAD

**Issue by ypapadop-amd** [ypapadop-amd/ggml#113]
These are to support the MNIST example.
